src/shared/components/ui/Modal/Modal.js

```diff
@@ -65,173 +65,188 @@ const cssClass = styles(scss)
+          <Spreader spread='tiny' />
+
+          {isClosable && <Close onClick={onClick} />}
+        </div>
+      </div>
+    )
 
-        <Spreader spread='tiny' />
+    const renderComponent = () => (
+      <div className={styles.modal__component}>
+        <div className={styles['modal__component--child']}>{component}</div>
 
         {isClosable && <Close onClick={onClick} />}
       </div>
-    </div>
-  )
+    )
 
-  const renderComponent = () => (
-    <div className={cssClass('modal__component')}>
-      <div className={cssClass('modal__component--child')}>{component}</div>
+    const defaultClassnames = Array.isArray(className)
+      ? className.map(className => {
+          return styles[className]
+            ? { [styles[className]]: true }
+            : { [className]: true }
+        })
+      : className
 
-      {isClosable && <Close onClick={onClick} />}
-    </div>
-  )
+    const modalStyles = useStyles({
+      ...defaultClassnames,
+      [styles['modal--fullscreen']]: size === 'fullscreen',
+      [styles['modal--big']]: size === 'big',
+      [styles['modal--medium']]: size === 'medium',
+      [styles['modal--center']]: isCentered,
+      [styles['modal--page']]: isPage
+    })
 
-  return (
-    <Fragment>
-      <PoseGroup animateOnMount flipMove={false}>
-        {isActive && (
-          <ModalAnimation key='ModalAnimation' className={scss.dialog}>
-            <div
-              className={cssClass(className, {
-                'modal--fullscreen': size === 'fullscreen',
-                'modal--big': size === 'big',
-                'modal--medium': size === 'medium',
-                'modal--center': isCentered,
-                'modal--page': isPage
-              })}
-            >
-              {isLoading ? (
-                <div className={scss.modal__body}>
-                  <Loader />
-                </div>
-              ) : (
-                <Fragment>
-                  {(isClosable || i18n.title || image || isEditable) &&
-                    (!isComponent ? renderTitle() : renderComponent())}
-                  {hasHeaderDivider && (
-                    <Fragment>
-                      <Spacer space='small' />
-                      <Divider />
-                    </Fragment>
-                  )}
-                  <div
-                    className={cssClass('modal__body', {
-                      'modal__body--has-footer': hasFooter
-                    })}
-                  >
-                    {disableOverflow ? (
-                      children
-                    ) : (
-                      <div style={overflowStyle}>{children}</div>
-                    )}
+    const bodyStyles = useStyles({
+      [styles['modal__body']]: true,
+      [styles['modal__body--has-footer']]: hasFooter
+    })
+
+    return (
+      <Fragment>
+        <PoseGroup animateOnMount flipMove={false}>
+          {isActive && (
+            <ModalAnimation key='ModalAnimation' className={styles.dialog}>
```

Comment:
To też, ale pamiętaj że jest potrzebny div z klasś styles.dialog. Raczej można zamienić `ModalAnimation` na div

Reply:
okej to usune react-pose z tego komponentu